### PR TITLE
Refactor PulpEntityContext

### PR DESCRIPTION
**Pull request by mdellweg**

### CODE REVIEW

File: pulpcore/cli/common/context.py [hunk 1]
```diff
@@ -71,6 +66,12 @@ class PulpContext:
     It is an abstraction layer for api access and output handling.
     """
 
+    def echo(self, message: str, nl: bool = True, err: bool = False) -> None:
+        print(message)  # we can do better...
+
+    def prompt(self, text: str, hide_input: bool = False) -> Any:
```
Comment (gerrod3) [inline pulpcore/cli/common/context.py:72]: Should these just be `NotImplementedError`?

Comment (gerrod3) [inline pulpcore/cli/common/context.py:72]: Should these just be `NotImplementedError`?

**Comment by gerrod3** [comment]
This work is for extracting out the shared logic between the CLI and ansible squeezer package?

File: pulpcore/cli/common/context.py (hunk 1 above)
Comment (mdellweg) [reply pulpcore/cli/common/context.py:72]: I dind't want to make it an abstract class.

**Review by ggainey — APPROVED** [review]
Looking good - thanks!

**Pull request merged**